control-plane/plugin/src/resources/node.rs

```diff
@@ -555,6 +555,49 @@ impl Drain for Node {
     }
 }
 
+fn allowed_topology_chars(key: char) -> bool {
+    key.is_ascii_alphanumeric() || matches!(key, '_' | '-' | '.')
+}
+fn validate_topology_key(key: &str) -> PluginResult {
+    if key.is_empty() {
+        return Err(Error::NodeLabelFormat {
+            error: "key must not be empty".to_string(),
+        });
+    }
+    if key.starts_with('/') || key.ends_with('/') {
+        return Err(Error::NodeLabelFormat {
+            error: "/ can only be used in the middle of the key".to_string(),
+        });
+    }
+
+    if key.chars().filter(|c| c == &'/').count() > 1 {
+        return Err(Error::NodeLabelFormat {
+            error: "/ is only allowed once".to_string(),
```

Comment:
"/ is only allowed once"
```suggestion
            error: "/ is allowed only once".to_string(),
```
Does this sound better? Maybe the above one as well?

Reply:
Both are the same to me but I can change to the latter if you prefer

Reply:
No, upto you, I think it's fine either way.

Reply:
I tweaked it a bit as I moved the errors into an enum